crypto/core_fetch.c

```diff
@@ -47,20 +71,18 @@ static void ossl_method_construct_this(OSSL_PROVIDER *provider,
      * of the passed method.
      */
 
-    if (data->force_store || !no_store) {
-        /*
-         * If we haven't been told not to store,
-         * add to the global store
-         */
-        data->mcm->put(data->libctx, NULL, method, data->operation_id,
-                       algo->algorithm_name,
-                       algo->property_definition, data->mcm_data);
+    if (global_store) {
+        /* If we haven't been told not to store, add to the global store */
+        (void)data->mcm->put(data->libctx, NULL, method, data->operation_id,
+                             algo->algorithm_name,
+                             algo->property_definition, data->mcm_data);
     }
 
-    data->mcm->put(data->libctx, data->store, method, data->operation_id,
-                   algo->algorithm_name, algo->property_definition,
-                   data->mcm_data);
+    (void)data->mcm->put(data->libctx, data->store, method, data->operation_id,
```

Comment:
I mean, should this be `} else {` ?

Reply:
No.  That would mean that `ossl_method_construct` will never find newly added methods.

Reply:
but this will be broken by this PR, since you skip that function when the
method is already in the global store, right?

Reply:
So unless ossl_method_construct executes under some kind of write_lock,
we can have a race condition making a second thread executing
the same ossl_method_construct fail.